### PR TITLE
`ExternalTaskSensor`: add `external_task_group_id` to `template_fields`

### PR DESCRIPTION
**Pull request by LiamOrmiston**
Add missing external_task_group_id parameter to the ExternalTaskSensor docstring and template_fields.

### CODE REVIEW

File: airflow/sensors/external_task.py [hunk 1]
```diff
@@ -121,7 +123,7 @@ class ExternalTaskSensor(BaseSensorOperator):
         or DAG does not exist (default value: False).
     """
 
-    template_fields = ["external_dag_id", "external_task_id", "external_task_ids"]
+    template_fields = ["external_dag_id", "external_task_id", "external_task_ids", "external_task_group_id"]
```
Comment (pankajastro) [inline airflow/sensors/external_task.py:128]:
```suggestion
    template_fields = ("external_dag_id", "external_task_id", "external_task_ids", "external_task_group_id")
```

**Review by pankajastro — APPROVED** [review]

File: airflow/sensors/external_task.py [hunk 2]
```diff
@@ -102,6 +102,8 @@ class ExternalTaskSensor(BaseSensorOperator):
         If ``None`` (default value) the sensor waits for the DAG. Either
         external_task_id or external_task_ids can be passed to
         ExternalTaskSensor, but not both.
+    :param external_task_group_id: The task_group_id that contains the task you want to
+        wait for.
```
Comment (uranusjr) [inline airflow/sensors/external_task.py]: Many operator classes add `(templated)` to templated fields and we should do that here as well.

File: airflow/sensors/external_task.py (hunk 1 above)
Comment (uranusjr) [reply airflow/sensors/external_task.py:128]: List works here as well, (Personally I prefer lists for a list of things.)

Comment (pankajastro) [reply airflow/sensors/external_task.py:128]: hmm, I prefer to use a tuple if an update does not require.  just curious why you prefer a list here particularly.

File: airflow/sensors/external_task.py (hunk 2 above)
Comment (LiamOrmiston) [reply airflow/sensors/external_task.py]: Went ahead and did this. Let me know if this is what you meant.

File: airflow/sensors/external_task.py (hunk 1 above)
Comment (eladkal) [reply airflow/sensors/external_task.py:128]: I think in most of the code base we have it as `template_fields: Sequence[str]`

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**